utils/s2n_mem.c

```diff
@@ -51,8 +53,8 @@ static int s2n_mem_init_impl(void)
     POSIX_ENSURE_GT(sysconf_rc, 0);
 
     /* page_size must be a valid uint32 */
-    POSIX_ENSURE_LTE(sysconf_rc, UINT32_MAX);
-
+    long max_page_size = MIN(UINT32_MAX, LONG_MAX);
+    POSIX_ENSURE_LTE(sysconf_rc, max_page_size);
```

Comment:
I might be missing something, but is this check doing anything differently than it was before? If UINT32_MAX is < LONG_MAX, `max_page_size` is UINT32_MAX, and this is the same check. If LONG_MAX is < UINT32_MAX, then this is just checking if `sysconf_rc` is < LONG_MAX, but `sysconf_rc` is already a long, so it's always < LONG_MAX, right?

Reply:
This was changed due to a complicated check as per [feedback comment](https://github.com/aws/s2n-tls/pull/4061#discussion_r1245799993). Although yes, `sysconf_rc` will always be less than `LONG_MAX`.

Reply:
Oh got it okay. The check isn't really needed if LONG_MAX < UINT32_MAX, but it avoids an if statement. That makes sense.